networkit/cpp/matching/test/MatcherGTest.cpp

```diff
@@ -279,4 +284,127 @@ TEST_F(MatcherGTest, testBSuitorMatcherDifferentB) {
     EXPECT_TRUE(M.isProper());
     EXPECT_FALSE(hasUnmatchedNeighbors(G, M));
 }
+
+TEST_F(MatcherGTest, testDynBSuitorInsertEdges) {
+    for (int i = 0; i < 10; i++) {
+        METISGraphReader graphReader;
+        auto G = graphReader.read("input/lesmis.graph");
+        std::vector<GraphEvent> events;
+        count m = 10;
+        // Select m edges of the graph, remove them but put them into edges for later insertion.
+        // This will make sure that the graph is valid.
+        for (count j = 0; j < m; j++) {
+            const auto [u, v] = GraphTools::randomEdge(G);
```

Comment:
Instead of testing on random insertions, can't we just delete all edges, insert one by one and test at each insertion?

Reply:
Good point, will refactor the test.